packages/cosmic-swingset/src/launch-chain.js

```diff
@@ -935,6 +951,17 @@ export async function launch({
           await doBootstrap(action);
         }
 
+        // The reboot-time upgradeSwingset() may have generated some
+        // remediation events that need to be injected at the right
+        // time (after catchup, before proposals). Push them onto
+        // runThisBlock before anything else goes there.
+        if (upgradeEvents) {
```

Comment:
Let's make sure this only happens in consensus restarts

```suggestion
        if (upgradeEvents) {
          isBootstrap ||
            upgradeDetails ||
            Fail`Unexpected upgrade events outside of consensus start`;
```